### ISSUE TITLE
Support custom link handler for links without a linktype.

### ISSUE THREAD

**Issue by danni** [wagtail/wagtail#3257]
A frequent request in sites is for external links to have the target=_blank attribute set on them. While for internal links we can set a link handler and `register_rich_text_link_handler`, you can do this for external links which don't have a `linktype` set.

Propose supporting a `None` link handler for links without `linktype` set which defaults to identity, but can be overridden as required.

**Comment by gasman** [comment]
👍  This would be my preferred approach to implementing #1167, as it means the handling of the `target` attribute can be kept outside of Wagtail core and implemented as a contrib / external module instead.

**Comment by thibaudcolas** [comment]
Just a note that @arifin4web has been working on this at Wagtail Space, with promising results. From what we tried it should then be very simple to implement things like #1167 server-side.

Relevant SO questions to update once all of this comes through:

- https://stackoverflow.com/questions/33300941/making-external-links-open-in-a-new-window-in-wagtail/34270862#34270862
- https://stackoverflow.com/questions/50042134/how-to-force-external-urls-inserted-in-the-wagtail-draftail-editor-to-open-on-th

**Comment by thibaudcolas** [comment]
This is now implemented thanks to #5159, to be released in Wagtail v2.5. The implementation introduces support for `external` and `email` as separate link types for which handlers can be registered with `features.register_link_type`. It doesn't have a catch-all for other links without linktypes – if people write their custom links, they should use a custom link type.

Here is an example of how it works:

```python
    from django.utils.html import escape
    from wagtail.core import hooks

    def external_link_handler(attrs):
        href = attrs["href"]
        return '<a href="%s" rel="nofollow">' % escape(href)

    @hooks.register('register_rich_text_features')
    def register_external_link(features):
        features.register_link_type('external', external_link_handler)
```

Docs: https://docs.wagtail.io/en/latest/advanced_topics/customisation/rich_text_internals.html#link-rewrite-handlers

**Comment by Mrfuyu** [comment]
For Wagtail 2.9 for me works:

`
#myapp/wagtail_hooks.py
from django.utils.html import escape
from wagtail.core import hooks
from wagtail.core.rich_text import LinkHandler

class NewTabExternalLinkHandler(LinkHandler):
    identifier = 'external'

    @classmethod
    def expand_db_attributes(cls, attrs):
        href = attrs["href"]
        return '<a href="%s" rel="nofollow" target="_blank">' % escape(href)

@hooks.register('register_rich_text_features')
def register_external_link(features):
    features.register_link_type(NewTabExternalLinkHandler)
`
